### ISSUE TITLE
Training vs Testing

### ISSUE THREAD

**Issue by harirehm** [rodrigopivi/Chatito#30]
Seems like there is a recent update in the IDE. And I see a new field `('training: '2', 'testing': '1')`. In the previous version, we only had to specify a numeric value in between the brackets and I suppose, they are the number of training samples that has to be generated. What is this `'testing'` field that's newly added?

According to [docs](https://github.com/rodrigopivi/Chatito/blob/master/spec.md), your total data generated will be training + testing data count. What is this testing data? I'm a bit confused. We are creating training file here and how is testing data pitching in? There is no proper definition in docs that explains the difference between training data and testing data.

> **Note:** I'm generating RASA NLU training data using the chatito IDE.

**Comment by rodrigopivi** [comment]
Hi @harirehm ,

Generating testing dataset is optional, so you can just pass the training entity argument and it will be ok if you just want to train your rasa nlu model. The problem with only having training data, is that you will have to evaluate the model manually. To make things easier for evaluation, you can generate a testing dataset too.

You can use the testing data to evaluate how accurate the model is with data it never trained with, this way you can compare the accuracy of your models, makes them easier to maintain and improve. As mentioned, is not required, but is important to be aware of the accuracy of a model so you can get to the highest accuracy. Sometimes just adding more and more data, can degrade performance of a model because of overfitting, this kind of issues can only be detected if you have a testing dataset.

**Comment by rodrigopivi** [comment]
will keep this issue open to remember to improve docs on this topic

**Comment by harirehm** [comment]
Clean!
